library/ssl_tls.c

```diff
@@ -3771,6 +3778,15 @@ static int ssl_tls13_session_save(const mbedtls_ssl_session *session,
     needed += 8; /* ticket_creation_time or ticket_reception_time */
 #endif
 
+#if defined(MBEDTLS_SSL_SRV_C)
+    if (session->endpoint == MBEDTLS_SSL_IS_SERVER) {
```

Comment:
Where in the serialised session do we capture that a session had `endpoint == MBEDTLS_SSL_IS_SERVER`? (i.e. to prevent it being reloaded unless `endpoint == MBEDTLS_SSL_IS_SERVER`?)

Reply:
It is in `ssl_session_save/load` that are common to TLS 1.2 and TLS 1.3.